nexus/db-queries/src/db/datastore/support_bundle.rs

```diff
@@ -371,30 +398,21 @@ impl DataStore {
                         bundles_reassigned: 0,
                     };
 
-                    // Exit a little early if there are no bundles to re-assign.
-                    //
-                    // This is a tiny optimization, but really, it means that
-                    // tests without Nexuses in their blueprints can succeed if
-                    // they also have no support bundles. In practice, this is
-                    // rare, but in our existing test framework, it's fairly
-                    // common.
-                    if bundles_to_reassign.is_empty() {
-                        return Ok(report);
+                    if !bundles_to_reassign.is_empty() {
+                        // Reassign bundles that haven't been marked "fully failed"
+                        // to ourselves, so we can free their storage if they have
+                        // been provisioned on a sled.
+                        report.bundles_reassigned =
+                            diesel::update(dsl::support_bundle)
+                                .filter(dsl::id.eq_any(bundles_to_reassign))
+                                .set(
+                                    dsl::assigned_nexus
+                                        .eq(our_nexus_id.into_untyped_uuid()),
+                                )
+                                .execute_async(conn)
+                                .await?;
```

Comment:
took me a moment to realize that the only diff here is that we changed the `if` from an early return to just making the `update` query conditional --- but i agree that this feels less convoluted than the early return, especially if we wanted to add additional code that runs after this query later.

Reply:
yeah it felt like "I didn't need an extra comment to explain an early return", because it's self-explanatory that we'd only want to update rows if there are rows to be updated